### PR TITLE
feat(storage): support s3 compatible object storage

### PR DESCRIPTION
**Pull request by wcy-fdu**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?
This PR supports s3 compatible object storage.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* Installation and deployment 
* Connector (sources & sinks)
* SQL commands, functions, and operators
* RisingWave cluster configuration changes
* Other (please specify in the release note below)

### Release note

Please create a release note for your changes. In the release note, focus on the impact on users, and mention the environment or conditions where the impact may occur.

### CODE REVIEW

**Comment by wcy-fdu** [comment]
Let's hold this PR after COS is supported.

**Comment by wcy-fdu** [comment]
Let's hold this PR after COS is supported.

File: src/object_store/src/object/s3.rs [hunk 1]
```diff
@@ -431,13 +432,18 @@ impl ObjectStore for S3ObjectStore {
...
+            return Ok(());
+        }
 
         // If needed, split given set into subsets of size with no more than `MAX_LEN` objects.
         for start_idx /* inclusive */ in (0..paths.len()).step_by(MAX_LEN) {
             let end_idx /* exclusive */ = cmp::min(paths.len(), start_idx + MAX_LEN);
             let slice = &paths[start_idx..end_idx];
-
-            // Create identifiers from paths.
+            //Create identifiers from paths.
```
Comment (hzxa21) [inline src/object_store/src/object/s3.rs:446]: nits: need an extra space

**Review by hzxa21 — APPROVED** [review]
LGTM. Thanks for the PR!

File: src/utils/runtime/src/lib.rs [hunk 2]
```diff
@@ -96,8 +95,7 @@ pub fn init_risingwave_logger(settings: LoggerSettings) {
         // Configure log output to stdout
         let fmt_layer = tracing_subscriber::fmt::layer()
             .compact()
-            .with_ansi(settings.colorful)
-            .with_timer(time::OffsetTime::local_rfc_3339().expect("could not get local offset!"));
+            .with_ansi(settings.colorful);
```
Comment (BugenZhao) [inline src/utils/runtime/src/lib.rs:98]: May I ask whether this reversion of #6038 is intended? If not, I'll add it back.

**Pull request merged**